arkouda/pdarrayIO.py

```diff
@@ -172,11 +172,13 @@ def read_parquet(filenames : Union[str, List[str]],
     """
     if isinstance(filenames, str):
         filenames = [filenames]
-    if isinstance(dsetname, str):
-        dsetname = [dsetname]
+    if datasets is None:
+        datasets = get_datasets_allow_errors(filenames, True) if allow_errors else get_datasets(filenames[0], True)
+    if isinstance(datasets, str):
+        datasets = [datasets]
```

Comment:
This is my first time looking at this code so I'm not sure but

Should we add something similar to the `else` block in `read_all` to make sure we throw a `ValueError` listing the missing datasets?
https://github.com/Bears-R-Us/arkouda/blob/883d7010ebd6464751c6e948f2914c48125d7fda/arkouda/pdarrayIO.py#L284-L294
It's not entirely clear to me what the `nonexistent`would look like since L289-290 should never run because it's identical to the check at L286-287. I'm not sure who the author of `read_all` is but maybe @reuster986 or @mhmerrill would know?

Reply:
@pierce314159, @reuster986 wrote the original code for this although I don't remember who touched it last... @bmcdonald3 @glitch ???

Reply:
I think you are right that 289-290 would never run in the current state, so that can likely be removed, but I think what nonexistent is checking is whether or not the datasets don't exist in the file.

For example, say you had a file with this schema: `{"col1": int_col, "col2": str_col}` and you call `ak.read_all("my-file", ['col1', 'bad-name'])` then nonexistent would be `set('bad-name')` because it is the set of the input datasets without the ones that actually exist in the file. It hits the `else` case because it is not just a string, but a list of datasets.

Admittedly, I did not think about this deeply when I was just copy/pasting the code over to the Parquet work, but now that I have looked at it more, I think that check should be moved outside of the else and should always run over the datasets to make sure no bad datasets slip by.

I'll update this PR to reflect that change; should I only change it in the Parquet code for this PR or should I do it in both places? (`read_all` and `read_parquet`)

Reply:
After rereading Pierce's original comment, I realize that I responded to a different question.

> Should we add something similar to the else block in read_all to make sure we throw a ValueError listing the missing datasets?

I think we should move the existing code that is in the else block that throws the value error outside of the else block to ensure it is run every time, but isn't that code in the else block already today? 

I think it was missing from the Parquet side, which I have now added.

Reply:
ohhh yeah that makes sense! It's @mhmerrill's call but I'm in favor of updating `read_all` as well

Reply:
oh yeah, I don't think I really knew what I was asking lol. I just figured the check probably needed to happen both places and was confused about the unreachable code in L289-290. Your explanation helped a lot

Reply:
can we merge something today and do another PR to resolve this?